jersey-servers/src/main/java/com/palantir/remoting2/servers/jersey/TraceEnrichingFilter.java

```diff
@@ -91,4 +94,23 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
             return Optional.of(header.equals("1"));
         }
     }
+
+    private static void addUserIdMetadataToSpan(ContainerRequestContext requestContext) {
+        Optional<String> userId = tryGetUnverifiedUserId(requestContext);
+        if (userId.isPresent()) {
+            Tracer.addSpanMetadata(ImmutableMap.of(BearerTokenLoggingFilter.USER_ID_KEY, userId.get()));
+        }
+    }
+
+    private static Optional<String> tryGetUnverifiedUserId(ContainerRequestContext requestContext) {
+        String userIdKey = BearerTokenLoggingFilter.getRequestPropertyKey(BearerTokenLoggingFilter.USER_ID_KEY);
+        Object maybeUserId = requestContext.getProperty(userIdKey);
```

Comment:
@uschi2000 - yeah (3) was the first approach, ruled out for duplicate processing
@gracew - I met with Rob and Dave and it was decided that we didn't want to rely on the `Priority` for now. Instead, this PR now just adds supporting adding metadata on spans. I will make a contour specific filter that composes the `BearerTokenLoggingFilter` and another of lower priority that creates a local span with the userId in the request context.